src/duckling/project/asset.service.ts

```diff
@@ -0,0 +1,47 @@
+import {Injectable} from '@angular/core';
+import {loader, Texture} from 'pixi.js';
+import {BehaviorSubject} from 'rxjs';
+
+import {StoreService} from '../state';
+
+export type AssetType = "TexturePNG" | "FontTTF" | "TextureJPEG";
+
+export interface Asset {
+    type : AssetType,
+    key : string
+}
+
+@Injectable()
+export class AssetService {
+    constructor(private _store : StoreService) {
+    }
+
+    private _assets : Asset[] = [];
+    numberOfAssets : BehaviorSubject<number> = new BehaviorSubject(0);

```

Comment:
I don't think we should expose a numberOfAssets observable. It should probably be something along the lines of assetLoaded.